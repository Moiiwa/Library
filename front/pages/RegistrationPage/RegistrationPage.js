import React from 'react'

import './RegistrationPage.css'

class RegistrationPage extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            user: {
                firstName: '',
                lastName: '',
                username: '',
                password: ''
            },
            submitted: false
        };

        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);

    }

    handleChange(event) {
        const { name, value } = event.target;
        const { user } = this.state;
        this.setState({
            user: {
                ...user,
                [name]: value
            }
        });
    }

    handleSubmit(event) {
        event.preventDefault();

        this.setState({ submitted: true });
        const { user } = this.state;
        if (user.firstName && user.lastName && user.username && user.password) {

        }
    }

    render() {

        const { registering } = this.props;
        const { user, submitted } = this.state;

        return (
            <div className="col-md-6 col-md-offset-3 jumbotron">
                <h1 className="display-4">Registration</h1>

                <form name="form" id="submit" onSubmit={this.handleSubmit}>
                    <div className={'f' + (submitted && !user.firstName ? ' has-error' : '')}>
                        <label htmlFor="firstName">First Name</label>
                        <input type="text" className="form-control" name="firstName" value={user.firstName} onChange={this.handleChange} />
                        {submitted && !user.firstName &&
                            <div className="help-block">First Name is required</div>
                        }
                    </div>

                    <div className={'form-group' + (submitted && !user.lastName ? ' has-error' : '')}>
                        <label htmlFor="lastName">Last Name</label>
                        <input type="text" className="form-control" name="lastName" value={user.lastName} onChange={this.handleChange} />
                        {submitted && !user.lastName &&
                            <div className="help-block">Last Name is required</div>
                        }
                    </div>

                    <div className={'form-group' + (submitted && !user.username ? ' has-error' : '')}>
                        <label htmlFor="username">Username</label>
                        <input type="text" className="form-control" id="username" name="username" value={user.username} onChange={this.handleChange} />
                        {submitted && !user.username &&
                            <div className="help-block">Username is required</div>
                        }
                    </div>

                    <div className={'form-group' + (submitted && !user.password ? ' has-error' : '')}>
                        <label htmlFor="password">Password</label>
                        <input type="password" className="form-control" name="password" value={user.password} onChange={this.handleChange} />
                        {submitted && !user.password &&
                            <div className="help-block">Password is required</div>
                        }
                    </div>

                    <div className="form-group">
                        <button className="btn btn-primary">Register</button>
                    </div>
                </form>

            </div>
        )
    }
}


export default RegistrationPage;