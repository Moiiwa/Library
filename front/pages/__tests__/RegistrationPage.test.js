import React from 'react';
import RegistrationPage from '../RegistrationPage/RegistrationPage';
import toJson from 'enzyme-to-json';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('RegistrationPage', () => {
    let wrapper;
    let mockSubmit;

    beforeEach(() => {
        mockSubmit = jest.fn();
        wrapper = shallow(<RegistrationPage submit={mockSubmit} />);
    });

    it('renders correctly', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("handle change", () => {
        const mockEvent = {
            target: {
                name: "username",
                value: "test"
            }
        };

        const expected = {
            user: {
                firstName: "",
                lastName: "",
                username: "test",
                password: "",
            },
            "submitted": false,
        };

        wrapper.instance().handleChange(mockEvent);
        expect(wrapper.state()).toEqual(expected)
    });

    it('handle submit', () => {
        const fakeEvent = { preventDefault: () => console.log('preventDefault') };

        const instance = wrapper.instance()
        const handleSubmitMock = jest.spyOn(instance, 'handleSubmit')
        instance.forceUpdate();

        const but = wrapper.find('#submit')
        but.simulate('submit', fakeEvent);
        expect(handleSubmitMock).toHaveBeenCalled()
    });

});