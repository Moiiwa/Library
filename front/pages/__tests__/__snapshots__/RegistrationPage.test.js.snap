// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationPage renders correctly 1`] = `
<div
  className="col-md-6 col-md-offset-3 jumbotron"
>
  <h1
    className="display-4"
  >
    Registration
  </h1>
  <form
    id="submit"
    name="form"
    onSubmit={[Function]}
  >
    <div
      className="f"
    >
      <label
        htmlFor="firstName"
      >
        First Name
      </label>
      <input
        className="form-control"
        name="firstName"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="lastName"
      >
        Last Name
      </label>
      <input
        className="form-control"
        name="lastName"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="username"
      >
        Username
      </label>
      <input
        className="form-control"
        id="username"
        name="username"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="form-control"
        name="password"
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <button
        className="btn btn-primary"
      >
        Register
      </button>
      <Link
        className="btn btn-link"
        replace={false}
        to="/login"
      >
        Cancel
      </Link>
    </div>
  </form>
</div>
`;
